### ISSUE TITLE
Send deleted repository resources to Artifactory trash when destroying

### ISSUE THREAD

**Issue by pshutt97** [jfrog/terraform-provider-artifactory#1094]
**Is your feature request related to a problem? Please describe.**
When running the destroy command or any command that affects the presence of resources not within the STATEFILE, the resource is immediately wiped from storage and the database. It's also not sent to the trash like other deletion processes currently do in Artifactory. This can cause possible irrecoverable data loss due to an incorrect command that is run.

**Describe the solution you'd like**
It would be ideal to mitigate possible impactful data loss by sending destroyed repository resources to the trash, along with their artifacts, instead of being completely wiped from the application. This way, in case of an accident with running Terraform, data can be restored if necessary.

**Comment by alexhung** [comment]
@pshutt97 Thanks for the suggestion. The current [Artifactory API for repository deletion](https://jfrog.com/help/r/jfrog-rest-apis/delete-repository) does not support moving repository to Trash.

I'll make a feature request internally to the Artifactory team. I encourage you to also do the same with your JFrog customer support contact.

**Comment by pshutt97** [comment]
> [@pshutt97](https://github.com/pshutt97) Thanks for the suggestion. The current [Artifactory API for repository deletion](https://jfrog.com/help/r/jfrog-rest-apis/delete-repository) does not support moving repository to Trash.
> 
> I'll make a feature request internally to the Artifactory team. I encourage you to also do the same with your JFrog customer support contact.

Thanks Alex! He had actually directed me to create this. Should I also open a support ticket as well?

**Comment by alexhung** [comment]
@pshutt97 Yes because this is not specifically a TF provider issue. This feature must be added to the Artifactory API first before I can add support for it in the provider.

**Comment by pshutt97** [comment]
Awesome, thanks!!!

**Comment by alexhung** [comment]
Related to #1069
